### PR TITLE
config/ApiVersion is read from dll and it value excluded from tests

### PR DESCRIPTION
**Pull request by JohannesFinsveen**

### CODE REVIEW

**Comment by JohannesFinsveen** [comment]
Used productVersion which includes the commit id 0.0.2-alpha.0.44+a1f513bb66b90ecc5459be00eda9fd11b60377ae
I did not see any  0.0.2-alpha.0.44, the other option was FileVersion which was "0.0.2.0"
As it is for human consumption I think i prefer not to fiddle with the string. Agree?

I also exclude the value from tests to avoid maintainence in the expected file.

**Comment by likp** [comment]
> Used productVersion which includes the commit id 0.0.2-alpha.0.44+a1f513bb66b90ecc5459be00eda9fd11b60377ae
> I did not see any 0.0.2-alpha.0.44, the other option was FileVersion which was "0.0.2.0"
> As it is for human consumption I think i prefer not to fiddle with the string. Agree?

Initially I thought that the FileVersion would be better, but giving it some thought I agree that the productVersion is better because then we are able to see if it is a beta or not which we can not see if we use FileVersion.

**Comment by runejo** [comment]
I found a solution to this and I have made a PR https://github.com/PxTools/PxApiSpecs/pull/104

- https://learn.microsoft.com/en-us/dotnet/core/compatibility/sdk/8.0/source-link
- https://stackoverflow.com/questions/77050814/fileversioninfo-productversion-suddenly-contains-git-commit-hash

File: PxWeb/Config/Api2/PxApiConfigurationOptions.cs [hunk 1]
```diff
@@ -1,10 +1,24 @@
...
-        public string ApiVersion { get; set; } = "2.0";
+        private readonly string _apiVersion;
+
+        public PxApiConfigurationOptions()
+        {
+            Assembly myLibAssembly = Assembly.Load("PxWeb.Api2.Server");
+            System.Diagnostics.FileVersionInfo fvi = System.Diagnostics.FileVersionInfo.GetVersionInfo(myLibAssembly.Location);
+
+            string specVersion = fvi.ProductVersion ?? "0.0";
+            _apiVersion = specVersion;
```
Comment (runejo) [inline PxWeb/Config/Api2/PxApiConfigurationOptions.cs:17]: I prefer implicit types and more verbose variable names, like this

```
var api2ServerAssembly = Assembly.Load("PxWeb.Api2.Server");
var fileVersionInfo = System.Diagnostics.FileVersionInfo.GetVersionInfo(api2ServerAssembly.Location);
_apiVersion = fileVersionInfo.ProductVersion ?? "2.0.0";
```

Comment (runejo) [reply PxWeb/Config/Api2/PxApiConfigurationOptions.cs:17]: and yeah, if for some reason it fails the default version should be `2.0.0` :-)

**Review by runejo — APPROVED** [review]

**Pull request merged**